test/functional/feature_remove_pruned_files_on_startup.py

```diff
@@ -18,38 +19,54 @@ def mine_batches(self, blocks):
         for _ in range(n):
             self.generate(self.nodes[0], 250)
         self.generate(self.nodes[0], blocks % 250)
-        self.sync_blocks()
 
     def run_test(self):
         blk0 = self.nodes[0].blocks_path / "blk00000.dat"
         rev0 = self.nodes[0].blocks_path / "rev00000.dat"
         blk1 = self.nodes[0].blocks_path / "blk00001.dat"
         rev1 = self.nodes[0].blocks_path / "rev00001.dat"
         self.mine_batches(800)
-        fo1 = os.open(blk0, os.O_RDONLY)
-        fo2 = os.open(rev1, os.O_RDONLY)
-        fd1 = os.fdopen(fo1)
-        fd2 = os.fdopen(fo2)
+
+        self.log.info("Open some files to check that this may delay deletion")
+        fd1 = open(blk0, "rb")
+        fd2 = open(rev1, "rb")
         self.nodes[0].pruneblockchain(600)
 
         # Windows systems will not remove files with an open fd
         if platform.system() != 'Windows':
-            assert not os.path.exists(blk0)
-            assert not os.path.exists(rev0)
-            assert not os.path.exists(blk1)
-            assert not os.path.exists(rev1)
+            assert not blk0.exists()
+            assert not rev0.exists()
+            assert not blk1.exists()
+            assert not rev1.exists()
         else:
-            assert os.path.exists(blk0)
-            assert not os.path.exists(rev0)
-            assert not os.path.exists(blk1)
-            assert os.path.exists(rev1)
+            assert blk0.exists()
+            assert not rev0.exists()
+            assert not blk1.exists()
+            assert rev1.exists()
 
-        # Check that the files are removed on restart once the fds are closed
+        self.log.info("Check that the files are removed on restart once the fds are closed")
         fd1.close()
         fd2.close()
         self.restart_node(0)
-        assert not os.path.exists(blk0)
-        assert not os.path.exists(rev1)
+        assert not blk0.exists()
+        assert not rev1.exists()
+
+        self.log.info("Check that a reindex will wipe all files")
+
+        def ls_files():
+            ls = [
+                entry.name
+                for entry in self.nodes[0].blocks_path.iterdir()
+                if entry.is_file() and any(map(entry.name.startswith, ["rev", "blk"]))
+            ]
+            return sorted(ls)
```

Comment:
Given that this pr was about `identical and requires less code`, it seems to me we can express this in a more compact form as well:
```suggestion
        ls_files = lambda: sorted(f.name for f in self.nodes[0].blocks_path.glob("*0*.dat"))
```

Reply:
> Also, I don't think your glob is identical, because it can match dat files other than rev/blk

Got it, what do you think about:
```python
ls_files = lambda: sorted(f.name for f in self.nodes[0].blocks_path.iterdir() if f.name.startswith(("rev", "blk")))
```

Reply:
> I am happy to drop it, if you want.

Of course not, what I want is maintainable code with high signal-to-noise ratio.
But if you don't think my alternative suggestions don't make the code maintainable, I'm not forcing you, I've already ACK-ed.